releases/branches.md

```diff
@@ -0,0 +1,20 @@
+# Branches
+
+This page describes branch conventions we follow.
+
+## Release long-lived branches
```

Comment:
Just a little driveby content structure review, hope you don't mind.

----

"Release long-lived branches" seems more like a call to action than a description of what is in this section. Perhaps it could be called "Release Branches"

Also I think the first sentence (we keep long-lived branches for...) deserves emphasis, it is the 'why', and you could explain why there are release lines that you still need to maintain so that people have a clearer context.

Also, we're defining a new domain term here "Release Branches", that can be long-lived. So maybe it should be "We keep release branches for..." instead of "We keep long-lived branches for..."

The branches are actually a communication tool, they say to everyone "Here's a fork of code that we still want to maintain, probably because it is still in production somewhere." These types of branches can be identified by the naming convention you are specifying below.

It's worth noting that there are other types of branches in Giant Swarm. Feature branches. Those currently don't have a naming convention, and that's fine. But it's nice to be obvious about what we're defining here:

1. What is a release branch - I suggest also adding here a 'why do we need release branches'
2. How we name and identify "release branches"
3. Rules around where tags can be depending on their format.